common/config/config.go

```diff
@@ -502,15 +507,21 @@ func (c *Config) fillDefaults() error {
 			c.Persistence.DataStores[k] = store
 		}
 	}
-	// filling RPCName with a default value if empty
-	if c.ClusterMetadata != nil {
-		for k, cluster := range c.ClusterMetadata.ClusterInformation {
-			if cluster.RPCName == "" {
-				cluster.RPCName = "cadence-frontend"
-				c.ClusterMetadata.ClusterInformation[k] = cluster
-			}
+
+	for name, cluster := range c.ClusterMetadata.ClusterInformation {
+		if cluster.RPCName == "" {
+			// filling RPCName with a default value if empty
+			cluster.RPCName = "cadence-frontend"
 		}
+		c.ClusterMetadata.ClusterInformation[name] = cluster
```

Comment:
nit: I guess we can move this assignment back, into the if.

Reply:
ohh right.